### PR TITLE
New briefing text and extendedText for the campaign

### PR DESCRIPTION
**Pull request by Thorneel**
WIP for the new campaign scenario:
 - All briefing texts replaced
 - All briefings (except the initial planet) have added extendedText, so a different text is displayed in the mission selection screen and in the in-game objective screen.

Codex pages have not been written yet, so this is only half of the actual work. The endings, in particular, are by necessity codex entries.
However, many players will probably not read most of the codex entries, if any, and this is how the story is initially conveyed, so it is still important that it can stand on its own.

English is not my native language, and this format with series of very short texts is unusual for me, so please watch out for typos, weird expressions and unclear sentences.
Feedback and (constructive) criticism on the scenario itself and the varied entries is also very welcome.

Tested on 1024x768 resolution for avoiding text overflow, and minimize scrolling in the objective/tips in-game menu.

Story inspired by the game Duskers (or at least the early part I played), memories from the beginning of the Sunrise campaign, summaries of previous Planetwars factions, and whatever the missions or the planet's position inspired me at the moment.

### CODE REVIEW

File: campaign/sample/planets/planet12.lua [hunk 1]
```diff
@@ -26,7 +26,11 @@ local function GetPlanet(planetUtilities, planetID)
 			primaryType = "G8V",
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24457",
-			text = [[The enemy is well entrenched but, luckily for you, they have a poorly defended outpost and you have a squad of Dominatrices. Capture the outpost to get a head start, then steal an army of Tanks and march on their main base.]]
+			text = "This rich industrial world is quite well-defended, but a dormant saboteur unit recognized my IFF and signaled me. I should be able to subvert those defences with its help."
+			.. "\n "
+			.. "\nAs far as I can tell, it identified me as allied to the rebels against whoever was the Empire controlling this world."
+			,
+			extendedText = "Hostile forces are well entrenched but, luckily for me, they have a poorly defended outpost and I now have a squad of Dominatrices. Captureing the outpost will give me a head start, then I can steal an army of Tanks and march on their main base."
```
Comment (Anarchid) [inline campaign/sample/planets/planet12.lua]: Capturing

File: campaign/sample/planets/planet2.lua [hunk 2]
```diff
@@ -26,7 +26,11 @@ local function GetPlanet(planetUtilities, planetID)
 			primaryType = "G4V",
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24417",
-			text = [[Your opponent has prepared powerful close-range Stardust turrets and Reavers to defeat your raiders. Build a Cloakbot Factory, then counter their strategy with longer-ranged Ronins to secure victory.]]
+			text = "Where is everyone? Automated systems and non-sapient life are still there, but no trace of intelligent beings. No humans, no cyborgs, no free machines, nothing..."
+			.. "\n "
+			.. "\nMost of the unit blueprints are gone. I will need to find new copies on my way out of here."
+			,
+			extendedText = "There are powerful close-range Stardust turrets and Reavers protecting them from my raiders. Once I have a Cloakbot Factory up and running, I will have to counter them with loger-ranged Ronins and open the way for taking their production systems out."
```
Comment (Anarchid) [inline campaign/sample/planets/planet2.lua]: loger -> longer

File: campaign/sample/planets/planet27.lua [hunk 3]
```diff
@@ -26,7 +26,11 @@ local function GetPlanet(planetUtilities, planetID)
 			primaryType = "G9V",
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24469",
-			text = [[Your ally can take care of any surface targets, but you'll need to support them with Claymore depth charge and Flail AA hovercraft against underwater and aerial threats.]]
+			text = "This planet was renowned for its tourism resorts and galaxy-famous culinary traditions - or so the advertisement broadcasts tell me. I don't remember ever hearing about it, but it is true this world has been relatively spared from collateral damage, compared to others in the sector."
+			.. "\n "
+			.. "\nUltimately, as the strategic picture beakem grimmer, even such a place suffered military occupation."
```
Comment (Anarchid) [inline campaign/sample/planets/planet27.lua]: beakem -> became

File: campaign/sample/planets/planet36.lua [hunk 4]
```diff
@@ -26,7 +26,11 @@ local function GetPlanet(planetUtilities, planetID)
 			primaryType = "G4V",
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24510",
-			text = [[Use the fast-moving Locust raider gunships to curtail your opponent's expansion, then Nimbus support gunships to finish them off.]]
+			text = "Occupation forces on this world have dwindled to a few dormant bots. Hopefully I can land and take them out before they awaken..."
+			.. "\n "
+			.. "\nAnother rebel planet that fell to a merciless imperial punitive expedition. I wonder what drove so many worlds to raise against the Empire."
```
Comment (Anarchid) [inline campaign/sample/planets/planet36.lua]: rise

File: campaign/sample/planets/planet49.lua [hunk 5]
```diff
@@ -25,7 +25,13 @@ local function GetPlanet(planetUtilities, planetID)
 			primaryType = "G6VI",
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24566",
-			text = [[The artillery cannons of your enemies' battleships are pretty big, but the Big Bertha plasma cannon is positively humongous. Use your static artillery to suppress the enemy until reinforcements arrive.]]
+			text = "There are two massive armies fighting down there, but both are declaring themselves Imperial forces. This didn't start with a glitch either, it was clearly two humans that initiated this. What could have driven them to do such a thing?"
+			.. "\n "
+			.. "\nI really don't want to fight both of them at the same time. Time to negociate with derelict automata again..."
```
Comment (Anarchid) [inline campaign/sample/planets/planet49.lua]: negotiate

File: campaign/sample/planets/planet56.lua [hunk 6]
```diff
@@ -25,7 +25,11 @@ local function GetPlanet(planetUtilities, planetID)
 			primaryType = "G4V",
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24510",
-			text = [[Many raiders have tried - and failed - to take the ancient technology on this planet by brute force. The automated defence system has decayed over time but a Detriment crushes any attempt to build a base. Stay on the move with Athena mobile constructors, building up an army to destroy the Detriment.]]
+			text = "There is technology on this world, that is both much more ancient and more andvanced than it should be. There are the remains of several raiding parties sent by the Empire to try - without success - and capture it."
+			.. "\n "
+			.. "\nAutomated defenses have suffered from the raids, but are still operational - and a Detriment roaming around. How can it be working after half an eternity exposed and without maintenance?"
+			,
+			extendedText = "That Detriment will crushe any attempt to build a base. I have to stay on the move with Athena mobile constructors, until I can build up an army strong enough to take it down."
```
Comment (Anarchid) [inline campaign/sample/planets/planet56.lua]: crush

File: campaign/sample/planets/planet59.lua [hunk 7]
```diff
@@ -25,7 +25,13 @@ local function GetPlanet(planetUtilities, planetID)
...
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24614",
-			text = [[Provoked by a rival salvager, the Artefact on this planet has released an automatic defensive measure - nanobots convert any destroyed unit wrecks into zombies, hostile to all intruders. Use the Ultimatum's Disintegrator Gun to put them all the way down, then get away before the Artefact does something even nastier.]]
+			text = "Rebel scientists were studying an artefact found in an archeological site, when they were overrun by the Empire. Desperate for any advantage they could, they activated it, releasing a zombie nanoplague."
+			.. "\n "
+			.. "\nAll it cost them was the lives of every single inhabitant on the planet."
+			.. "\n "
+			.. "\nAnd now that thing is powering up again."
+			,
+			extendedText = "The plague will convert any destroyed unit wreck into zombies, hostile to all. I can the Ultimatum's Disintegrator Gun to put them all the way down, then deactivate that damn Artefact before it releases something even nastier."
```
Comment (Anarchid) [inline campaign/sample/planets/planet59.lua]: I can the -> i can use the

File: campaign/sample/planets/planet6.lua [hunk 8]
```diff
@@ -26,7 +26,13 @@ local function GetPlanet(planetUtilities, planetID)
...
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24429",
-			text = [[This battle is taking place at high altitude, so deploy Wind Generators for cheap and efficient energy income. Use the Knight's lightning gun to eliminate the enemy Jumpbots.]]
+			text = "This world is covered with islands, with a surprising variety of defences. Was it used for combat testing?"
+			.. "\n "
+			.. "\nMany of those islands seem isolated from each-other. I will only need to clear one, hopefully it won't contain any nasty experimental unit."
+			,
+			extendedText = "This battle is taking place at high altitude, I can use Wind Generators for cheap and efficient energy income."
+			.. "\n "
+			.. "\nThose Jumpbots I am up against seem rather strange. Not sure whta the best tactics are against them, but I hope my Knight's lightning gun will help."
```
Comment (Anarchid) [inline campaign/sample/planets/planet6.lua]: Whta

File: campaign/sample/planets/planet61.lua [hunk 9]
```diff
@@ -25,7 +25,11 @@ local function GetPlanet(planetUtilities, planetID)
 			primaryType = "F9IV",
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24614",
-			text = [[The high mountains of this battlefield, and the enemy's defensive bastions seated upon them, have so far resisted assault. The Funnelweb strider is well equipped to dislodge them.]]
+			text = "With the Empire growing increasingly unstable, they fell back to their home sectors, letting the periphery slowly decay into lawlessness. Hidden behind their fortress worlds, they hoped to regrow their strength and strike again, reconquering the galaxy once and for all."
+			.. "\n "
+			.. "\nWas it the last delusion of a decadent regime? Or would they have truly risen stronger in due time?"
+			,
+			extendedText = "Even with those backdoors, much of the local forces are still hostile - I will have to dislodge them from their mounatinous defensive bastions. Fortunately, those Funnelweb striders are well equiped for that particular task."
```
Comment (Anarchid) [inline campaign/sample/planets/planet61.lua]: mountainous

**Comment by Thorneel** [comment]
Thanks Anarchid, fixed the typos

File: campaign/sample/planets/planet53.lua [hunk 10]
```diff
@@ -26,7 +26,11 @@ local function GetPlanet(planetUtilities, planetID)
 			primaryType = "K2V",
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24510",
-			text = [[Your allies are struggling against cloaked skirmishers and artillery attacking them at range. Use Firewalker napalm artillery and jumping Jacks to find and destroy the hidden enemies.]]
+			text = "Local forces are emitting a general distress call. A rogue commander was trying to take this world over and carve herself a nice little kingdom - with her army still at it. Loosing control of the military? The Empire must have been crumbling from the inside."
```
Comment (Anarchid) [inline campaign/sample/planets/planet53.lua]: Loosing -> losing

File: campaign/sample/planets/planet64.lua [hunk 11]
```diff
@@ -25,7 +25,19 @@ local function GetPlanet(planetUtilities, planetID)
...
 			milRating = 1,
 			feedbackLink = "http://zero-k.info/Forum/Thread/24642",
-			text = [[This battle will take place on a massive salt plain, littered with sharp spikes. Use Paladin striders to dominate all aspects of the fight.]]
+			text = "Planet Intrepid, crown jewel of the Empire, garden of the East - this dead, barren world?"
+			.. "\n "
+			.. "\nI can see it now. Left unsupervised, maintenance failed, gardens dried and the entire ecosystem fell into a death spiral."
+			.. "\n "
+			.. "\nLeaving nothing but grey sand, haunted by decaying armies and fading memories."
+			,
+			extendedText = "This battle takes place on what was once a shallow lake, renown across the entire galaxy for its splendor."
```
Comment (Anarchid) [inline campaign/sample/planets/planet64.lua]: renowned

**Comment by GoogleFrog** [comment]
How old is your fork? I suspect the conflicts are whitespace changes from the 16th of August.

**Comment by Thorneel** [comment]
Five months old, so it checks out.
Merged from upstream/master added that to the pull request

Also fixed more typos

**Comment by Thorneel** [comment]
Finally got rid of those conflicts

**Comment by GoogleFrog** [comment]
I like it, although the text does not fit on the UI for smaller screens.or 1360x768.

image

I'll fiddle with the placement of elements for small windows.

**Comment by GoogleFrog** [comment]
In case you're unfamiliar with github, that message means I committed changes to your fork. You should pull and take a look at them to make sure they work for all planets. The main change is that I removed Primary and Type on small screens. If they are required they can be added somewhere else.

**Comment by Thorneel** [comment]
Primary and type not being visible on smaller resolutions is not a problem.
I didn't use primary, and type was mostly an inspiration/limitation to keep texts coherent - but types are visible with the in-game map anyway. It was useful to confirm that "crystalline" was unique for that one planet. So they are nice to have, but it should work fine without them.

Checking all the briefing texts on 1360x768, they are fine. Without primary and type, they even have extra space compared to 800 height. Thanks for fixing my oversight - getting those texts even shorter would have been a challenge!

In-game text adds a bit more scrolling for the longest ones, but it doesn't make a significant difference. 32 pixels is only about half the height of a tip.
I was pretty sure to have set it to 768 height, but somehow it was 800 all along...

**Pull request merged**